sdk/core/Azure.Core/src/ClientUtilities.cs

```diff
@@ -1,44 +1,29 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-#if AZURE_NULLABLE
-#nullable enable
-#endif
-
 using System;
 using System.Collections;
 using System.Collections.Generic;
-using System.Diagnostics.CodeAnalysis;
 
 namespace Azure.Core
 {
     /// <summary>
-    /// Argument validation.
+    /// Utility class for standardizing behavior across Azure SDK clients.  This type is intended for use
+    /// by developers building Azure SDK service clients.
+    ///
+    /// Its primary functionality is standardizing exception messages thrown from client methods.
+    /// When calling ClientUtilities.Assert to throw an exception, please make sure to include documentation
+    /// on the public methods indicating which exceptions are thrown.
     /// </summary>
-    /// <remarks>
-    ///   <para>This class should be shared via source using Azure.Core.props and contain only common argument validation.
-    ///     It is declared partial so that you can use the same familiar class name but extend it with project-specific validation.
-    ///     To extend the functionality of this class, just declare your own partial <see cref="Argument"/> class with project-specific methods.
-    ///   </para>
-    ///   <para>
-    ///     Be sure to document exceptions thrown by these methods on your public methods.
-    ///   </para>
-    /// </remarks>
-    internal static partial class Argument
+    public static class ClientUtilities
     {
-        // TODO: Add nullability attributes as needed when possible.
-
         /// <summary>
         /// Throws if <paramref name="value"/> is null.
```

Comment:
TODO: improve docs.

Might be good to add cut-and-paste samples of refdocs for callers to include in their method docs when calling.  We could add analyzers to ensure it's done correctly?